pkg/kubelet/network/cni/cni.go

```diff
@@ -43,6 +44,8 @@ type cniNetworkPlugin struct {
 	host           network.Host
 	execer         utilexec.Interface
 	nsenterPath    string
+	mu             sync.Mutex
+	podMutexes     map[string]*sync.Mutex
```

Comment:
Any time I see a map of mutexes, I am alarmed.  Who is cleaning this up?  It's sort of a code-smell.  In this case I can't see anything OBVIOUSLY wrong.  Could use some great comments here, though.  Explain what this is for and that it only holds pods that have not been torn down yet.